### DISCUSSION TITLE
Optional file upload validation

### DISCUSSION THREAD

**Discussion by nickngqs** [adonisjs/core#1723] [Help] [answered]
I'm doing validation on the schema. How do I validate a file only if a file is uploaded. I tried using the optional chaining.

```
const validationSchema = schema.create({
	image: schema.file.optional({
		size: '10mb',
		extnames: ['jpg', 'png', 'jpeg'],
	})
})

const productDetails = await request.validate({
	schema: validationSchema,
	existsStrict: true,
})
```

**Comment by thetutlage** [comment, marked as answer]
Isn't the `file.optional` not working? And how did you verify that is not working?

**Reply by nickngqs** [reply]
I think i made a mistake, sorry! closing the issue
